modules/clients.py

```diff
@@ -27,50 +27,41 @@ def done_at(self, time):
 
 class Clients(object):
 
-    def __init__(self, env, server, requests, process_time=0.00001, create_request_rate=0.01, max_requests=float("inf")):
+    def __init__(self, env, server, requests, process_time=0.00001, create_request_rate=0.01, max_requests=float("inf"), service_time=0.0035, memory=0.02):
         self.env = env
         self.server = server
         self.requests = requests
         self.process_time = process_time
         self.queue = simpy.Store(env)  # the queue of requests
         self.action = env.process(self.send_request())
-        self.create_request = env.process(self.create_request(create_request_rate, max_requests))
+        self.create_request = env.process(self.create_request(create_request_rate, max_requests, service_time, memory))
 
     def send_request(self):
         while True:
             if len(self.queue.items) > 0:  # check if there is any request to be processed
                 request = yield self.queue.get()  # get a request from store
-                print("At %.3f, CLIENTS a client sent a request" % self.env.now)
                 yield self.env.process(self.server.request_arrived(request))
 
             yield self.env.timeout(self.process_time) # wait for...
 
-    def create_request(self, create_request_rate, max_requests):
-        """ Create requests """
+    def create_request(self, create_request_rate, max_requests, service_time, memory):
         count = 1
         while count <= max_requests:
-            duration = 0.035
-            memory = 0.02
-            request = Request(self.env.now, duration, memory, self)
-            print("At %.3f, CLIENTS a client made a request" % self.env.now)
+            request = Request(self.env.now, service_time, memory, self)
             yield self.queue.put(request)
             yield self.env.timeout(create_request_rate)
```

Comment:
Strange to use a ratio to timeout. For instance, rate is in req/sec, for instance .. timeout is in seconds. The units are different.

Am I missing something?

Reply:
The idea was to put a delay between each request creating. For send 10 request for second, we should set create_request_rate with the value of 0.1. It would mean that after each 0.1 seconds, one request is created.

Reply:
@danielfireman do you think that a good name for this is "create_request_time"?

Reply:
I would say this is simply the `delay_between_requests` which could be calculated from a ratio. For instance, If one would like to send 10 requests within each second of the experiment, the `delay_between_requests` would be 100 milliseconds. But please notice that the latter is not a ratio (the units are not coherent).

Reply:
I get it. @danielfireman sounds good to you keep a ratio instead of the time between the creations? For that, we should keep the ratio of creating request and get that time internally.